File: PlayTen/PlayTen.DAL/Entities/Training.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace PlayTen.DAL.Entities
{
    public class Training
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public DateTime DateStart { get; set; }
        public DateTime DateEnd { get; set; }
        public int NumberOfParticipants { get; set; }
        public string Description { get; set; }
        public bool HasBalls { get; set; }
        [Required]
        public int PlaceId { get; set; }
        public Place Place { get; set; }
        [Required]
        public string OwnerId { get; set; }
        public User User { get; set; }
        public ICollection<Participant> Participants { get; set; }
        
        
    }
}

File: PlayTen-Client/playten/src/app/models/training.ts
import { Place } from "./place";
import { Participant } from "./participant";

export class Training {
 public training!: TrainingInfo;
 public isUserTrainingOwner!: boolean;
 public isUserParticipant!: boolean;
 public isUserApprovedParticipant!: boolean;
 public isUserUndeterminedParticipant!: boolean;
 public isUserRejectedParticipant!: boolean;
 public isTrainingFinished!: boolean;
}

export class TrainingInfo {
 public trainingId?: number;
 public name!: string;
 public dateStart!: Date;
 public dateEnd!: Date;
 public description!: string;
 public placeId!: number;
 public numberOfParticipants!: number;
 public hasBalls!: boolean;
 public ownerId!: string;
 public place?: Place;
 public participants?: Participant[];
}
